src/BlazorWasmSdk/Tasks/ComputeBlazorPublishAssets.cs

```diff
@@ -549,24 +552,58 @@ private void ApplyPublishProperties(ITaskItem newAsemblyAsset)
                     {
                         var finalCulture = !string.IsNullOrEmpty(culture) ? culture : inferredCulture;
                         var assemblyName = Path.GetFileName(candidate.GetMetadata("RelativePath").Replace("\\", "/"));
-                        satelliteAssemblyToPublish.Add((finalCulture, assemblyName), candidate);
+                        if (!satelliteAssemblyToPublish.ContainsKey((finalCulture, assemblyName)))
+                        {
+                            satelliteAssemblyToPublish.Add((finalCulture, assemblyName), candidate);
+                        }
+                        else
+                        {
+                            Log.LogMessage("Duplicate candidate '{0}' found in ResolvedFilesToPublish", candidate.ItemSpec);
```

Comment:
```suggestion
                            Log.LogMessage("Duplicate candidate '{0}' found in SatelliteAssemblyToPublish", candidate.ItemSpec);
```

Reply:
candidate is referring to the list of resolved files to publish, am I missing something here?

Reply:
I may be misunderstanding this, but the current error message says a duplicate is found in `ResolvedFilesToPublish`, however the check on line 555 is for `satelliteAssemblyToPublish` (ie. not `resolvedAssemblyToPublish`). I thought the error message may need to reflect the satellite rather than the resolved files.